src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -51,8 +57,24 @@ StartRemoteTransactionBegin(struct MultiConnection *connection)
 	 * side might have been changed, and that would cause problematic
 	 * behaviour.
 	 */
-	if (!SendRemoteCommand(connection,
-						   "BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED"))
+	beginCommand = "BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED";
+
+	/*
+	 * Append BEGIN and assign_distributed_transaction_id() statements into a single command
+	 * and send both in one step. The reason is purely performance, we don't want
+	 * seperate roundtrips for these two statements.
+	 */
+	appendStringInfo(beginAndSetDistributedTransactionId, "%s;", beginCommand);
+
```

Comment:
`beginCommand` as a separate string seems fairly useless here (and you're removing a const, because "" literals are const, but `beginCommand` is not). Just put into the `appendStringInfo()` call?